### PR TITLE
Fix formatting error on member chain

### LINKED ISSUE #15912: TypeError: Cannot read properties of undefined (reading 'type')
**Prettier 3.2.0**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuE8DOMAUBKAOlXGAOnRgCYd9CS4MKYALASygHMB+ItCAWzkZatsAbnwgANCAgAHGE2hpkoAIYAnVRADuABTUJFKZQBtNygJ6LJAI1XKwAa34BlZXwAyLOMgBmxtHGtbB2dpO0FkGFUAVwCQfx4mCOjYuAAPaThVJj5YYwAVTKg1JlofP1i0QSM4AEUoiHgyo39JACs0VKcq2vrGpF9m2IBHXrhtDWkDEGU0AFooODgAE2WJVFsmI0EAYV4eZWRpoyM1yrZqgEEYSKYrKPhtTI8FppaQBhgeIwB1ZnRQsBwJz6JhyABuoLMhzAaEsIDBMQAklAVrAnGAsrILiinDAzNVXrFpBp-N9bNJDsTaJkwV5JCx-KoYONlKx9oTJKFVIzDniMmgMUxZGtiSwYN8mEtGMgABwABkkqjgIyYSpZbIO-XKkhgyisEqlDGQZEkUX8eT1BgGbzgPCsyxWSzcyjYUVZcAAYhBVPtruFDPcICAAL7BoA)

```sh
--parser typescript
```

**Input:**

```tsx
test()
	.test2()
	.test2(thing?.something);

```

**Output:**

```tsx
TypeError: Cannot read properties of undefined (reading 'type')
    at Jn (https://prettier.io/lib/plugins/estree.js:7:16595)
    at be (https://prettier.io/lib/plugins/estree.js:9:605)
    at be (https://prettier.io/lib/plugins/estree.js:10:289)
    at https://prettier.io/lib/plugins/estree.js:36:14282
    at Array.every (<anonymous>)
    at https://prettier.io/lib/plugins/estree.js:36:14273
    at Array.some (<anonymous>)
    at Oi (https://prettier.io/lib/plugins/estree.js:36:14252)
    at Kr (https://prettier.io/lib/plugins/estree.js:36:14852)
    at Ta (https://prettier.io/lib/plugins/estree.js:36:58423)
```

**Expected output:**

Not exactly sure, it's an error.

**Why?**

This was not an error in 3.1.x.

### PR DESCRIPTION
**Pull request by sosukesuzuki**
## Description

Fixes #15912

## Checklist

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [ ] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by sosukesuzuki** [comment]
Maybe this is caused by https://github.com/prettier/prettier/pull/15806

File: src/language-js/utils/index.js [hunk 1]
```diff
@@ -405,6 +405,16 @@ function isTestCall(node, parent) {
...
+    // @ts-expect-error
+    if (node.object == null) {
+      // @ts-expect-error
+      node.object = node.expression.object;
+    }
+    // @ts-expect-error
+    if (node.property == null) {
+      // @ts-expect-error
+      node.property = node.expression.property;
+    }
```
Comment (sosukesuzuki) [inline src/language-js/utils/index.js:417]: @fisker This isn't a good. I'm merging this PR as a temporary solution, but do you have any better solution?

Comment (fisker) [reply src/language-js/utils/index.js]: Yes, this is wrong, should update logic where it's accessing the object/property.

Comment (fisker) [reply src/language-js/utils/index.js]: This may also cause `computed`/`shorthand` check part result wrong..

**Pull request merged**